pkg/apis/azure/validation/infrastructure.go

```diff
@@ -92,6 +92,20 @@ func ValidateInfrastructureConfig(infra *apisazure.InfrastructureConfig, nodesCI
 	if !infra.Zoned && infra.Networks.NatGateway != nil {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("networks", "natGateway"), infra.Networks.NatGateway, "NatGateway is currently only supported for zoned cluster"))
 	}
+	if infra.Zoned && infra.Networks.NatGateway != nil {
+		if !infra.Networks.NatGateway.Enabled && (len(infra.Networks.NatGateway.IPAddresses) > 0 || len(infra.Networks.NatGateway.IPAddressRanges) > 0) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("networks", "natGateway"), infra.Networks.NatGateway, "NatGateway is not enabled but ip addresses or ip ranges are specified"))
+		} else {
+			ipAddressesField := fldPath.Child("networks", "natGateway", "ipAddresses")
+			for i, ip := range infra.Networks.NatGateway.IPAddresses {
+				allErrs = append(allErrs, validateAzResourceReference(&ip, ipAddressesField.Index(i))...)
```

Comment:
```suggestion
				allErrs = append(allErrs, validateAzResourceReference(ip.DeepCopy(), ipAddressesField.Index(i))...)
```

Reply:
Why do we need a copy of the ip here instead of passing the reference?

Reply:
`ip` is the iteration variable for the `range`, and it always has the same reference

Reply:
ahh damn always fell on it... Thanks.

Reply:
done